### ISSUE TITLE
Syntax error not detected referencing class when targeting ES6

### ISSUE THREAD

**Issue by willbar** [Microsoft/TypeScript#11455]
**TypeScript Version:**  2.0.3

**Code**

```ts
var p = new Foo();
class Foo() {
}
```

**Expected behavior:**
Transpile error when targeting ES6. Because class declarations aren't hoisted this isn't valid and will fail at runtime.

**Actual behavior:** 
No transpile error.

**Comment by mhegazy** [comment]
looks like there is something else going on. i do not see this locally:

```cmd
c:\test\sandbox8>type a.tsx
var p = new Foo();
class Foo() {
}

c:\test\sandbox8>\tsc --v
Version 2.0.3

c:\test\sandbox8>tsc --t es6 a.tsx
a.tsx(2,10): error TS1005: '{' expected.
a.tsx(2,13): error TS1005: '=>' expected.

```

**Comment by wallverb** [comment]
@mhegazy remove `()`  from `class Foo()`

**Comment by yortus** [comment]
I can repro this, indeed no error is raised but it should be a static error.

The compiler *does* check class declaration order in other cases, like this:

```ts
class Derived extends Base {} // ERROR class must be declared after base class
class Base {};
```

**Comment by falsandtru** [comment]
See #5207.

**Comment by RyanCavanaugh** [comment]
Thanks @falsandtru
